couchbase/admin/collection/v1/collection.proto

```diff
@@ -28,7 +28,10 @@ message ListCollectionsRequest { string bucket_name = 1; }
 message ListCollectionsResponse {
   message Collection {
     string name = 1;
-    optional uint32 max_expiry_secs = 2;
+    oneof max_expiry_secs {
+      uint32 secs = 2;
+      bool no_expiry = 4;
```

Comment:
boolean seems like a strange field type. Would it make sense for this to be an enum, with values like "same as bucket" and "never expire"?

Reply:
I think we can avoid using an enum, I've left it as a bool but change the meaning of it, so that it covers the "same as bucket" and "never expire" cases, instead of just the latter. Let me know your thoughts.

Reply:
What do the different boolean values mean?

I'll try lobbying one more time for an enum. The advantages of a 2-element enum over a boolean:

* You can name the cases, like `max_expiry_special=SAME_AS_BUCKET` and `max_expiry_special=NEVER_EXPIRE`.

* You can easily add a third case in the future, like a hypothetical `max_expiry_special=END_OF_MONTH`. (That particular example might sound silly, but originally we didn't expect to have 2 special expiry types.

Reply:
- I'll admit no_expiry is unclear, but with the boolean named inherit_expiry I think it's fairly clear what the boolean values mean.
- I think possible future special values are unlikely, supporting no expiry is a pretty predictable special case, so choosing an ENUM just for that reason seems unnecessary
- Even if there were new special cases using a bool here doesn't stop us supporting them, it just makes it neater until we do (if we ever need to)